lib/treat-locks/npm.lock.js

```diff
@@ -23,21 +23,20 @@ const commonLock = require('./common.lock');
 const NpmOptions = require('./npm.options');
 
 function treatFields(dependencies, npmOptions = new NpmOptions()) {
-  if (npmOptions.replacePackageLockRegistry === true) {
-    if (npmOptions.registry === undefined) {
-      throw Error('replacePackageLockRegistry flag defined but registry');
-    }
+  if (npmOptions.replacePackageLockRegistry === true && npmOptions.registry === undefined) {
+    throw Error('replacePackageLockRegistry flag defined but registry is undefined');
   }
   Object.keys(dependencies).forEach((prop) => {
     const dependency = dependencies[prop];
-    if (npmOptions.replacePackageLockRegistry === true
+    if (typeof dependency === 'object'
+      && npmOptions.replacePackageLockRegistry === true
       && npmOptions.registry !== undefined
       && dependency.resolved !== undefined) {
       dependency.resolved = commonLock.replaceHost(dependency.resolved, npmOptions.registry);
     } else {
       delete dependency.resolved;
     }
-    if (npmOptions.skipIntegrity !== true) {
+    if (npmOptions.skipIntegrity === false) {
```

Comment:
it could even be `if (!npmOptions.skipIntegrity)` but up to you

Reply:
I kept it as false for consistency, as we use a strict check here `npmOptions.replacePackageLockRegistry === true` - I also prefer the readbility.